### DISCUSSION TITLE
[Proposal] Allow (customizable) profile pages on rubygems.org for gem authors

### DISCUSSION THREAD

**Discussion by rubyFeedback** [rubygems/rubygems.org#4199] [Ideas]
Hey there rubygems.org maintainers,

Have a look at a (semi-random) github page such as the one from kojix2 from Japan:

https://github.com/kojix2

On such an author's homepage, or central page, the author can describe a few
things, not just his projects, but also policies and what not. This issue is about
the latter, that is, policy descriptions.

I would like to propose that rubygems.org also allows for a profile page, similar
to github.

We currently have a profile such as:

https://rubygems.org/profiles/foobar
https://rubygems.org/profiles/andy_maleh # such as for Andy's page

We can not add any content to this though.

I would like to propose that we get the ability to have a LIMITED text-area to
add content. Let's say 500 characters or some such, to keep it small and not
burden the infrastructure of rubygems.org too much. By default this is
empty and not shown. Once added, a simple "edit" button can be shown,
and a "save" button or "update" button to add content to it.

This can then be shown as a link such as:

https://rubygems.org/profiles/andy_maleh/policies

or

https://rubygems.org/profiles/andy_maleh/description

or status or any other name that you deem appropriate.

Or, alternatively, it could be shown at the profiles, in a small
div-tag, on the left or perhaps right side. Or perhaps above or
below the gem listing.

I say the limit 500 that is rather arbitrary, but it can be reduced
or increased. So perhaps it can be lower.

What is the rationale and purpose of this?

Well - just as in github profile pages, we can mention things.

Specifically I would like to have this to be able to add some more
information about my own policies, e. g. when I deprecate and
remove old gems (it's fine if rubygems.org hosts them; I want
to focus mostly on recent code, for most of my gems, to keep the
maintenance load low).

That profile pages can also be used for other things, but my primary
objective is as a per-policy page. You can allow or disallow HREF links
how you want to - personally I think it may be better to keep it as
unrestricted as possible, but I understand if you want to keep it
low, to avoid problems (e. g. github has more resources, so maintaining
things is easier for them than for rubygems.org, I get that).

Anyway this is the proposal. Please feel free to close the issue at any
moment in time.

**Comment by simi** [comment]
I like the idea of extending user profile. Would you mind to comment on https://github.com/rubygems/rfcs/pull/51?
